Lib/asyncio/selector_events.py

```diff
@@ -695,12 +711,14 @@ def get_write_buffer_size(self):
 class _SelectorSocketTransport(_SelectorTransport):
 
     _start_tls_compatible = True
+    _sendfile_compatible = constants._SendfileMode.NATIVE
 
     def __init__(self, loop, sock, protocol, waiter=None,
                  extra=None, server=None):
         super().__init__(loop, sock, protocol, extra, server)
         self._eof = False
         self._paused = False
+        self._empty_waiter = None
```

Comment:
`_empty_waiter` should handle a case when it exists and a connection is aborted/closed. In which case we want to cancel it.

Reply:
This case is still isn't handled.

Reply:
Cancelling is confusing.
I prefer another exception -- and OSError or derived type is explicitly set

Reply:
Whatever, just make sure that if something awaits on `_empty_writer` does not wait *forever* because we don't cancel the future.